nixos/modules/services/databases/mysql.nix

```diff
@@ -252,10 +286,30 @@ in
 
   config = mkIf config.services.mysql.enable {
 
+    warnings = optional (cfg.extraOptions != null) "services.mysql.`extraOptions` is deprecated, please use services.mysql.`settings`.";
+
     services.mysql.dataDir =
       mkDefault (if versionAtLeast config.system.stateVersion "17.09" then "/var/lib/mysql"
                  else "/var/mysql");
 
+    services.mysql.plugins = optional (cfg.ensureUsers != []) "auth_socket.so";
+
+    services.mysql.settings.mysqld = mkMerge [
+      {
+        basedir = "${cfg.package}";
```

Comment:
`basedir` wasn't set previously, so is this here needed? Also fyi this should be the same as just `basedir = cfg.package`

Reply:
```
error: The option value `services.mysql.settings.mysqld.basedir' in `/home/andersea/nixpkgs/nixos/modules/services/databases/mysql.nix' is not of type `boolean or signed integer or string'.
(use '--show-trace' to show detailed location information)
```

Yeah I should remove that as it is something I will be adding to a future PR. Thanks for catching.

Reply:
Oh, never mind then, I guess it would need a `types.package` in the `oneOf` listing for it to work. Though one could argue that the `str` type is at fault for not accepting string-coercible types.